pkg/devfile/header.go

```diff
@@ -30,4 +30,24 @@ type DevfileMetadata struct {
 	// Map of implementation-dependant free-form YAML attributes.
 	// +optional
 	Attributes attributes.Attributes `json:"attributes,omitempty"`
+
+	// Optional devfile display name
+	// +optional
+	DisplayName string `json:"displayName,omitempty"`
+
+	// Optional devfile description
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	// Optional devfile tags
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+
+	// Optional devfile icon
+	// +optional
+	Icon string `json:"icon,omitempty"`
+
+	// Optional devfile global memory limit
+	// +optional
+	GlobalMemoryLimit string `json:"globalMemoryLimit,omitempty"`
```

Comment:
I'm not sure if `GlobalMemoryLimit` is needed (but I'm also not opposed to adding it). The story behind this field is that we needed a way to get the total memory required by a workspace, and this was hard to do from a very simple registry -- plugin memory requirements are stored in their own meta.yamls and can change arbitrarily while the devfile registry is running, and further the devfile itself can override the values set in the plugin. This field is *manually* set by the devfile author to smooth a use-case in Che, but the value here can drift apart from reality (if e.g. one of your plugins updates and starts using more memory). 

I wonder if the new devfile registry could offer some metrics/calculation to get this number directly without user input?

Reply:
Thanks for bringing that up, AFAIK currently the new devfile registry/OCI devfile registry doesn't provide a mechanism to directly get the number of global memory limit, so for the timing being, let's still add this field. Given this PR is only for purely migrating fields from meta.yaml, I think we can merge this PR for now, then open separate issue to discuss the existence of the specific field if it's needed as I think that may or may not bring more discussion and design.